src/Core/AuthenticationHelpers/EasyAuthAuthenticationBuilderExtensions.cs

```diff
@@ -27,20 +27,65 @@ public static AuthenticationBuilder AddEasyAuthAuthentication(
             throw new System.ArgumentNullException(nameof(builder));
         }
 
-        builder.AddScheme<EasyAuthAuthenticationOptions, EasyAuthAuthenticationHandler>(
-            authenticationScheme: EasyAuthAuthenticationDefaults.AUTHENTICATIONSCHEME,
-            displayName: EasyAuthAuthenticationDefaults.AUTHENTICATIONSCHEME,
-            options =>
-            {
-                if (easyAuthAuthenticationProvider is EasyAuthType.StaticWebApps)
+        if (easyAuthAuthenticationProvider is EasyAuthType.StaticWebApps)
+        {
+            //builder.Services.TryAddEnumerable(ServiceDescriptor.Singleton<IConfigureOptions<EasyAuthAuthenticationOptions>, EasyAuthConfigureOptions>());
+
+            builder.AddScheme<EasyAuthAuthenticationOptions, EasyAuthAuthenticationHandler>(
+                authenticationScheme: EasyAuthAuthenticationDefaults.SWAAUTHSCHEME,
+                displayName: EasyAuthAuthenticationDefaults.SWAAUTHSCHEME,
+                options =>
                 {
                     options.EasyAuthProvider = EasyAuthType.StaticWebApps;
-                }
-                else if (easyAuthAuthenticationProvider is EasyAuthType.AppService)
+                });
+        }
+        else
+        {
+            builder.AddScheme<EasyAuthAuthenticationOptions, EasyAuthAuthenticationHandler>(
+                authenticationScheme: EasyAuthAuthenticationDefaults.APPSERVICEAUTHSCHEME,
+                displayName: EasyAuthAuthenticationDefaults.APPSERVICEAUTHSCHEME,
+                options =>
                 {
                     options.EasyAuthProvider = EasyAuthType.AppService;
-                }
-            });
+                });
+        }
+
+        return builder;
+    }
+
+    /// <summary>
+    /// Used for ConfigureAuthenticationV2() where all EasyAuth schemes are registered.
+    /// This function doesn't register EasyAuthType.AppService if the AppService environment is not detected.
+    /// </summary>
+    /// <exception cref="System.ArgumentNullException"></exception>
+    public static AuthenticationBuilder AddEnvDetectedEasyAuth(this AuthenticationBuilder builder)
+    {
+        if (builder is null)
+        {
+            throw new ArgumentNullException(nameof(builder));
+        }
+
+        builder.AddScheme<EasyAuthAuthenticationOptions, EasyAuthAuthenticationHandler>(
+           authenticationScheme: EasyAuthAuthenticationDefaults.SWAAUTHSCHEME,
+           displayName: EasyAuthAuthenticationDefaults.SWAAUTHSCHEME,
+           options =>
+           {
+               options.EasyAuthProvider = EasyAuthType.StaticWebApps;
+           });
+
+        bool appServiceEnvironmentDetected = AppServiceAuthenticationInfo.AreExpectedAppServiceEnvVarsPresent();
```

Comment:
should we log if appServiceEnvironment is detected or not? could be useful for debugging.

Reply:
I'm adding logging for case when appservice IS detected, so that it is obvious we are registering the scheme. Doesn't make sense to log when it isn't detected since this would log even when SWA is used.